src/main/java/frc/robot/subsystems/Feeder.java

```diff
@@ -13,18 +14,33 @@
 import frc.robot.utils.diag.DiagColorSensor;
 import frc.robot.utils.smartshuffleboard.SmartShuffleboard;
 
+import java.util.ArrayList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+
 public class Feeder extends SubsystemBase {
 
     private final WPI_TalonSRX feederMotor;
     private final I2C.Port i2cPort = I2C.Port.kMXP;
     private final ColorSensor colorSensor;
+    private final BlockingQueue<ColorMatchResult> colorReadingBuffer = new LinkedBlockingQueue<>();
```

Comment:
An alternative solution that does not involve a queue is to use an AtomicInteger that represents the Maximum certainty or an AtomicBoolean that represents a find. This way you don't consume memory and do not block.

Another alternative is to store just the raw reading rather than the matched one, since the matching can take a lot of compute, so we can do the matching only when needed.

Reply:
I like the idea of an AtomicIntereger.  I feel like storing the raw reading and then having to match the color in the main thread would slow things down, although that is just a hunch. My original idea was to use an atomic boolean however I did not like the idea of having an AtomicBoolean because we have different confidence thresholds for incoming and exiting feeder.

Reply:
The Eason for delaying the color matching is to do it once per cycle rather than every time we read the value.
We can do that later